### PR TITLE
Private and public access modes for ANDI

### PR DESCRIPTION
**Pull request by bbrewer-forge**

### CODE REVIEW

File: apps/andi/lib/andi_web/controllers/edit_controller.ex [hunk 1]
```diff
@@ -14,7 +14,7 @@ defmodule AndiWeb.EditController do
         |> render("404.html")
 
       dataset ->
-        if is_curator do
+        if is_curator and Andi.private_access?() do
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi_web/controllers/edit_controller.ex:17]: Not happy about all the mixing of concerns we have to do here, but the JSR is not there to change this dramatically

File: apps/andi/lib/andi_web/controllers/edit_controller.ex [hunk 2]
```diff
@@ -50,4 +60,14 @@ defmodule AndiWeb.EditController do
         live_render(conn, AndiWeb.EditOrganizationLiveView, session: %{"organization" => org, "is_curator" => is_curator})
     end
   end
+
+  def access_level(conn) do
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi_web/controllers/edit_controller.ex:64]: Oops - this is cruft

File: apps/andi/lib/andi_web/live/dataset_live_view/table.ex [hunk 3]
```diff
@@ -44,4 +44,7 @@ defmodule AndiWeb.DatasetLiveView.Table do
 
   defp get_status_class(nil), do: "unset"
   defp get_status_class(status), do: String.downcase(status)
+
+  defp edit_type(false), do: "submissions"
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi_web/live/dataset_live_view/table.ex:48]: I just shotgunned this in as an implementation. Let me know if you'd like it to suck less

File: apps/andi/test/integration/andi_web/live/data_dictionary_field_editor_test.exs [hunk 4]
```diff
@@ -158,7 +159,7 @@ defmodule AndiWeb.EditLiveView.DataDictionaryFieldEditorTest do
 
   describe "certain fields in the data dictionary editor are unavaliable for non curator users" do
     setup %{curator_subject: curator_subject, public_subject: public_subject} do
-      {:ok, public_user} = Andi.Schemas.User.create_or_update(public_subject, %{email: "bob@example.com"})
+      {:ok, public_user} = Andi.Schemas.User.create_or_update(public_subject, %{email: "bob162@example.com"})
```
Comment (bbrewer-forge) [inline apps/andi/test/integration/andi_web/live/data_dictionary_field_editor_test.exs:162]: Oops - was using these to track down a leaky database sandbox case. I can revert if desired.

File: apps/andi/lib/andi_web/auth/ensure_access_level_for_route.ex [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    else
+      {:excluded, true} ->
+        conn
+
+      _ ->
+        not_found(conn)
+    end
+  rescue
+    e ->
+      message = "Unexpecgted error occurred while checking route " <> inspect(e)
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi_web/auth/ensure_access_level_for_route.ex:36]: oops

File: apps/andi/lib/andi_web/auth/ensure_access_level_for_route.ex [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    end
+  end
+
+  def call(conn, opts) do
+    router = Keyword.fetch!(opts, :router)
+    exclusions = Keyword.get(opts, :exclusions, [])
+
+    with %{plug: plug, plug_opts: plug_opts} <- Phoenix.Router.route_info(router, conn.method, conn.path_info, conn.host),
+         {controller, action} = resolve_controllers(plug, plug_opts),
+         {:excluded, false} <- {:excluded, plug in exclusions},
```
Comment (bbrewer-forge) [inline apps/andi/lib/andi_web/auth/ensure_access_level_for_route.ex:23]: oops, that should be controller

**Review by cameronmarsh — APPROVED** [review]

**Comment by cameronmarsh** [comment]
seems good to me

**Pull request merged**